src/graph/query_graph.c

```diff
@@ -145,22 +145,14 @@ void _BuildQueryGraphAddEdge(const GraphContext *gc,
     Node *src = QueryGraph_GetNodeByAlias(qg, src_node->alias);
     Node *dest = QueryGraph_GetNodeByAlias(qg, dest_node->alias);
     Edge *e = Edge_New(src, dest, edge->ge.label, edge->ge.alias);
-
-    // Get relation matrix.
+    
+    //Set edge relation ID.
     if(edge->ge.label == NULL) {
         Edge_SetRelationID(e, GRAPH_NO_RELATION);
-        e->mat = Graph_GetAdjacencyMatrix(g);
     } else {
         Schema *s = GraphContext_GetSchema(gc, edge->ge.label, SCHEMA_EDGE);
-        if(s) {
-            Edge_SetRelationID(e, s->id);
-            e->mat = Graph_GetRelationMatrix(g, s->id);
```

Comment:
:+1: This removal is great.

Reply:
That's exactly where we gain the performance.